addons/purchase/models/purchase.py

```diff
@@ -125,6 +125,7 @@ def _compute_invoice(self):
     product_id = fields.Many2one('product.product', related='order_line.product_id', string='Product', readonly=False)
     user_id = fields.Many2one('res.users', string='Purchase Representative', index=True, tracking=True, default=lambda self: self.env.user)
     company_id = fields.Many2one('res.company', 'Company', required=True, index=True, states=READONLY_STATES, default=lambda self: self.env.company_id.id)
+    currency_rate = fields.Float("Currency Rate", compute='_compute_currency_rate', compute_sudo=True, store=True, digits=(12, 6), readonly=True, help='The rate of the currency to the currency of rate 1 applicable at the date of the order')
```

Comment:
not sure you have to specify digits for it.

Reply:
help='Ratio between the purchase order currency and the company currency'

Reply:
@amoyaux  We put digits here because default decimal precision is 3 digit ... to make bigger rounding we have to put 6 digit.